src/components/BottomWidgetsContainer.jsx

```diff
@@ -0,0 +1,63 @@
+import React       from 'react';
+import PropTypes   from 'prop-types';
+import { connect } from '../store/Connect';
+
+import '../../sass/components/_bottom-widget-container.scss';
+
+
+class BottomWidgetsContainer extends React.Component {
+    componentDidUpdate() {
+        if (React.Children.count(this.props.children)) {
+            this.props.updateChartMargin(200);
+        } else {
+            this.props.updateChartMargin(50);
+        }
+    }
+    render() {
+        const {
+            bottom,
+            isReadyToShow,
+            children,
+            top,
+        } =  this.props;
+
+        if (!isReadyToShow) {
+            return null;
+        }
+
+        const styles = {
+            top,
+            bottom,
+        };
+
+        return (
+            <div
+                className="cq-bottom-ui-widgets"
+                style={styles}
+            >
+                { children }
+            </div>
+        );
+    }
+}
+
+BottomWidgetsContainer.propTypes = {
```

Comment:
@morteza-binary we are not using prop-type? why you need it here?

Reply:
@MahboobehMohammadi It should be added to all the project to make the code more clear.